generators/app/templates/src/app/components/AppNavigator/index.js

```diff
@@ -13,18 +13,18 @@ const AppNavigator = () => {
   const state = useSelector(state => state.nav);
   const dispatch = useDispatch();
 
-  const onBackPress = () => {
-    if (state.index === 0) return false;
-    dispatch(NavigationActions.back());
-    return true;
-  };
-
   useEffect(
     () => {
+      const onBackPress = () => {
```

Comment:
I don't think this is necessary. Is there a reason why you made this change? Suposedly, each time the component is rendered again, `onBackPress` will be regenerated using the new state.index and dispatch, you shouldn't need to have it inside the useEffect for that to work.

Reply:
This change is for the reasons stated in these documents: https://es.reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies and https://react-redux.js.org/next/api/hooks#usedispatch...The rules of hooks also recommend it.

Reply:
That's nor what I meant. In the docs it says that 
```
function Example() {
  function doSomething() {
    console.log(someProp);
  }

  useEffect(() => {
    doSomething();
  }, []); // 🔴
}
```

Isn't safe because you're not passing `doSomething` as a dependency like this:

```
function Example() {
  function doSomething() {
    console.log(someProp);
  }

  useEffect(() => {
    doSomething();
  }, [doSomething]); 
}
```